genromfs.c

```diff
@@ -961,7 +969,6 @@ int processdir(int level, const char *base, const char *dirname, struct stat *sb
 			}
 		}
 	}
-	if (dirfd) closedir(dirfd);
```

Comment:
needs to be replaced with `free(namelist)` and the individual `namelist[...]` entries need to get freed as well.